src/Redux/Subscriptions/thunks.js

```diff
@@ -0,0 +1,84 @@
+import {
+  putSubscriptionSuccess,
+  putSubscriptionPending,
+  putSubscriptionError,
+  deleteSubscriptionSuccess,
+  deleteSubscriptionPending,
+  deleteSubscriptionError
+} from './actions';
+
+import { handleDisplayToast, setContentToast } from '../Shared/ResponseToast/actions';
+
+export const putSubscription = (
+  newSubscription,
+  subscriptionById,
+  subscriptions,
+  setSubscriptions,
+  id
+) => {
+  return async (dispatch) => {
+    dispatch(putSubscriptionPending());
+    try {
+      const isoDate = newSubscription.date ? new Date(newSubscription.date).toISOString() : '';
+      const body = JSON.stringify({
+        classes: newSubscription.classes,
+        members: newSubscription.members,
+        date: isoDate
+      });
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/subscription/${id}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: body
+      });
+      const data = await response.json();
+      if (response.status === 200) {
+        const updatedSubscriptions = subscriptions.map((sub) => {
```

Comment:
line 37-45 are needed? , i mean, where are you using the subscription saved on the set?
Can be done on a global variable using redux?